http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/dao/HttpSamDAO.scala

```diff
@@ -423,5 +425,5 @@ object ResourceTypeName {
 }
 
 final case object NotFoundException extends NoStackTrace
-final case class AuthProviderException(traceId: TraceId, msg: String)
+final case class AuthProviderException(traceId: TraceId, msg: String, statusCode: StatusCode)
```

Comment:
Needs something like 
```
final case class AuthProviderException(traceId: TraceId, msg: String, override val statusCode: StatusCode)
  extends LeoException(message = s"${traceId} | AuthProvider error: $msg")
```
I guess

Reply:
yeah I didn't bother compiling, but I guess thats flying too close to the sun